presto-main/src/main/java/com/facebook/presto/operator/PagesIndex.java

```diff
@@ -315,8 +315,8 @@ public PagesHashStrategy createPagesHashStrategy(List<Integer> joinChannels, Opt
     public PagesHashStrategy createPagesHashStrategy(List<Integer> joinChannels, Optional<Integer> hashChannel, Optional<JoinFilterFunction> joinFilterFunction)
     {
         try {
-            return joinCompiler.compilePagesHashStrategyFactory(types, joinChannels)
-                    .createPagesHashStrategy(ImmutableList.copyOf(channels), hashChannel);
+            return joinCompiler.compilePagesHashStrategyFactory(types, joinChannels, joinFilterFunction.map(f -> f.getClass()))
```

Comment:
Issue 1:

This function has only one callsite, and `joinFilterFunction` is always `Optional.empty()`. Therefore, inline this function into `createPagesHashStrategy(List<Integer> joinChannels, Optional<Integer> hashChannel)`.

Issue 2:
* For compiled `LookupSource`, we delegate to `JoinFilterFunction` directly. In generated `PagesHashStrategy`, `hasFilterFunction` always return false and `applyFilterFunction` always returns true.
  * But if you look at `JoinCompiler.generateXxxFilterFunction`, this not apparent at all.
* For interpreted `InMemoryLookupSource`, we delegate to `SimplePagesHashStrategy.applyFilterFunction`, which delegates to `JoinFilterFunction`.

This is confusing and hard to reason about because there are 2 possible places that have the logic to evaluate JFF. And you always have it in exactly one of the 2 places, either PagesHashStrategy or InMemoryLookupSource. Let's remove JFF from PagesHashStrategy entirely and have `InMemoryLookupSource` call `JoinFilterFunction` directly.